holoviews/plotting/bokeh/renderer.py

```diff
@@ -114,7 +119,21 @@ def get_size(self_or_cls, plot):
 
         Returns a tuple of (width, height) in pixels.
         """
-        return (plot.state.height, plot.state.height)
+        if not isinstance(plot, Model):
+            plot = plot.state
+        if isinstance(plot, (Row, Column, ToolbarBox, WidgetBox)):
+            if not plot.children: return 0, 0
+            horizontal = isinstance(plot, Row) or (isinstance(plot, ToolbarBox) and plot.toolbar_location not in ['right', 'left'])
+            widths, heights = zip(*[self_or_cls.get_size(child) for child in plot.children])
+            w_agg, h_agg = (np.sum, np.max) if horizontal else (np.max, np.sum)
+            width, height = w_agg(widths), h_agg(heights)
+        elif isinstance(plot, (Chart, Figure)):
+            width, height = plot.plot_width, plot.plot_height
+        elif isinstance(plot, (Plot, Div, DataTable)):
+            width, height = plot.width, plot.height
+        if width is None: width = 0
```

Comment:
Certain models do not declare a width/height and rather than guess at it I'd take the hit and just say it's zero. The only thing that I've found that's actually affected by this is the ``Div`` model used to add titles to grid plots.